pkgs/development/haskell-modules/lib.nix

```diff
@@ -76,6 +76,14 @@ rec {
     fixupPhase = ":";
   });
 
+  # link executables statically against haskell libs to reduce closure size
+  justStaticExecutables = drv: overrideCabal drv (drv: {
+    enableSharedExecutables = false;
+    isLibrary = false;
+    doHaddock = false;
+    postFixup = "rm -rf $out/lib $out/nix-support $out/share/doc";
```

Comment:
@bennofs to be honest, I just used what was already there. I don't know the reasoning behind the doc removal, but we could probably check with existing code what is left behind it that folder isn't deleted. If haddock is removed, I don't see no harm deleting also the whole doc folder.